source/patterns/@aws-solutions-constructs/aws-lambda-kinesisstream/README.md

```diff
@@ -0,0 +1,115 @@
+
+Typescript
+``` typescript
+import { Construct } from 'constructs';
+import { Stack, StackProps } from 'aws-cdk-lib';
+import { LambdaToKinesisStreamProps } from '@aws-solutions-constructs/aws-lambda-kinesisstream';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+
+new LambdaToKinesisStream(this, 'LambdaToKinesisStream', {
+  lambdaFunctionProps: {
+    runtime: lambda.Runtime.NODEJS_18_X,
+    handler: 'index.handler',
+    code: lambda.Code.fromAsset(`lambda`)
+  }
+});
+```
+
+Python
+``` python
+from aws_solutions_constructs.aws_lambda_kinesis_stream import LambdaToKinesisStream
+from aws_cdk import (
+    aws_lambda as _lambda,
+    aws_kinesis as kinesis,
+    Stack
+)
+from constructs import Construct
+
+LambdaToKinesisStream(self, 'LambdaToKinesisStream',
+                        lambda_function_props=_lambda.FunctionProps(
+                          runtime=_lambda.Runtime.PYTHON_3_9,
+                          handler='index.handler',
+                          code=_lambda.Code.from_asset('lambda')
+                        )
+                      )
+```
+
+Java
+``` java
+import software.constructs.Construct;
+
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awscdk.services.lambda.*;
+import software.amazon.awscdk.services.lambda.eventsources.*;
+import software.amazon.awscdk.services.lambda.Runtime;
+import software.amazon.awsconstructs.services.lambdakinesisstream.*;
+
+new LambdaToKinesisStream(this, "LambdaToKinesisStream", new LambdaToKinesisStreamProps.Builder()
+        .lambdaFunctionProps(new FunctionProps.Builder()
+                .runtime(Runtime.NODEJS_18_X)
+                .code(Code.fromAsset("lambda"))
+                .handler("index.handler")
+                .build())
+        .build());
+```
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|existingLambdaObj?|[`lambda.Function`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Function.html)|Existing instance of a Lambda Function object, providing both this and `lambdaFunctionProps` will cause an error.|
+|lambdaFunctionProps?|[`lambda.FunctionProps`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.FunctionProps.html)|User provided props to override the default props for the Lambda Function.|
+|existingStreamObj?|[`kinesis.Stream`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_kinesis.Stream.html)|Existing instance of a Kinesis Data Stream, providing both this and `kinesisStreamProps` will cause an error.|
+|kinesisStreamProps?|[`kinesis.StreamProps`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_kinesis.StreamProps.html)|Optional user-provided props to override the default props for the Kinesis Data Stream.|
+|createCloudWatchAlarms|`boolean`|Whether to create recommended CloudWatch Alarms (defaults to true).|
```

Comment:
This construct needs a vpc option - the Lambda function can make the call from an EIP in a VPC to an Gateway for KinesisStreams in the VPC (look at lambda-sqs for an example).

Reply:
Done.